libsolidity/formal/ModelCheckerSettings.h

```diff
@@ -26,6 +26,29 @@
 namespace solidity::frontend
 {
 
+struct ModelCheckerContracts
+{
+	/// By default all contracts are analyzed.
+	static ModelCheckerContracts All() { return {}; }
+
+	static std::optional<ModelCheckerContracts> fromString(std::string const& _contracts);
+
+	bool all() const { return contracts.empty(); }
+
+	bool has(std::string const& _source) const { return contracts.count(_source); }
```

Comment:
Does it makes sense to add a check if `all()` and return true in both of these?
That way we do not need to call `all() || has(...)`

Reply:
It could simplify the usage logic, but I feel like in that case the function would be doing too much, since `ModelCheckerContracts` doesn't really know the engine logic